internal/commands/vaultsecrets/integrations/list.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package integrations
+
+import (
+	"context"
+	"fmt"
+
+	preview_secret_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/models"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-06-13/client/secret_service"
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/flagvalue"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/heredoc"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+type ListOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	Output  *format.Outputter
+	IO      iostreams.IOStreams
+
+	Type          string
+	Client        secret_service.ClientService
+	PreviewClient preview_secret_service.ClientService
+}
+
+func NewCmdList(ctx *cmd.Context, runF func(*ListOpts) error) *cmd.Command {
+	opts := &ListOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		IO:            ctx.IO,
+		Output:        ctx.Output,
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "list",
+		ShortHelp: "List Vault Secrets integrations.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets integrations list" }} command lists Vault Secrets generic integrations.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `List twilio integrations:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets integrations list --type "twilio"
+				`),
+			},
+			{
+				Preamble: `List all generic integrations:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets integrations list"
+				`),
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "type",
+					DisplayValue: "TYPE",
+					Description:  "The optional type of integration to list.",
+					Value:        flagvalue.Simple("", &opts.Type),
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			if runF != nil {
+				return runF(opts)
+			}
+			return listRun(opts)
+		},
+	}
+	return cmd
+}
+
+func listRun(opts *ListOpts) error {
+	if opts.Type == "" {
+		var integrations []*models.Secrets20231128Integration
+		params := &preview_secret_service.ListIntegrationsParams{
+			Context:        opts.Ctx,
+			ProjectID:      opts.Profile.ProjectID,
+			OrganizationID: opts.Profile.OrganizationID,
+		}
+		for {
+			resp, err := opts.PreviewClient.ListIntegrations(params, nil)
```

Comment:
Not sure why I'm getting a 404 when hitting `ListIntegrations`. Even when trying directly by hitting the API. Lmk if anyone has ideas!
<img width="951" alt="Screenshot 2024-07-30 at 4 35 54 PM" src="https://github.com/user-attachments/assets/20968e00-f2a8-4504-8d6d-becb8afb5527">
<img width="1529" alt="Screenshot 2024-07-30 at 4 36 08 PM" src="https://github.com/user-attachments/assets/57fb0117-6b92-4a80-afdf-466d81ab2b7b">

Reply:
Nothing helpful to add :( it seems like you're doing the right thing, and the endpoint _should_ be querying from the third_party_integrations table

Reply:
Just tested this locally and it seems to worked for me, happy to help debug this if its still an issue

Reply:
Not sure what the issue is here, I'll take some time to investigate deeper when I get a bit more time in the project. But for now I was able to confirm the general list works against my personal environment, and Anton was able to confirm it worked against prod for him (checked out this branch).